### PR TITLE
Item search

### PR DESCRIPTION
**Pull request by bkingsley727**
Item Search
Scroll to Menu Item in restaurant
Fixed missing info on Restaurant Header
Added Open in Maps

### CODE REVIEW

File: lib/pages/homePage.dart [hunk 1]
```diff
@@ -177,7 +180,10 @@ class _HomePageState extends State<HomePage> {
 
   void getSearch(String s) async {
     var restList = await SearchUtil.restaurantByGeoAndName(currentLocation, s);
+    var itemList = await SearchUtil.menuItemByGeoAndName(currentLocation, 20, s);
+
```
Comment (carteralbrecht) [inline lib/pages/homePage.dart:184]: Could we only do the work for whatever tab the user is currently on? Doing both up front is very expensive

File: lib/pages/homePage.dart [hunk 2]
```diff
@@ -193,6 +199,23 @@ class _HomePageState extends State<HomePage> {
       );
     }
 
+    for (MenuItem i in itemList) {
+      Restaurant r = await Documenu.getRestaurant(i.restaurantId);
+      var avg = await _fsm.getDocData(_fsm.menuItemCollection, i.id, "avgRating");
+      var dist = await GeoUtil.distanceToRestaurant(currentLocation, r);
```
Comment (carteralbrecht) [inline lib/pages/homePage.dart:205]: If we had a GeoUtil.distanceToMenuItem, could we avoid making the `await Documenu.getRestaurant()` call right now? What fields of `r` are actually needed at this time?

**Review by carteralbrecht — CHANGES_REQUESTED** [review]
I think we can make it faster

File: lib/components/menuItemSearchTile.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+              Expanded(
+                child: Text(
+                  widget.restaurant.name,
+                  style: TextStyle(fontSize: 12, fontStyle: FontStyle.italic, color: global.accentGrayDark,),
+                  maxLines: 1,
+                  overflow: TextOverflow.ellipsis,
+                ),
+              ),
+              Container(
+                child: Text(" • " + widget.milesAway.toStringAsFixed(2))
```
Comment (carteralbrecht) [inline lib/components/menuItemSearchTile.dart:61]: Screen Shot 2021-03-29 at 7 16 46 PM

Should this say miles away? or have a unit of some sort

File: lib/pages/homePage.dart (hunk 2 above)
Comment (bkingsley727) [reply lib/pages/homePage.dart:205]: We need all of r in order to build the route for the restaurant page for the onTap() method. I can redesign the card if we don't wat some of the other info but I believe there is o way to combine any of the calls currently

**Review by carteralbrecht — APPROVED** [review]

**Pull request merged**